Sources/TensorFlow/Core/ShapedArray.swift

```diff
@@ -995,3 +995,14 @@ fileprivate extension _ShapedArrayProtocol where Scalar: Equatable {
         }
     }
 }
+
+// conditional conformance of array to module and layer
+extension Array: Module where Element: Layer, Element.Input == Element.Output {
```

Comment:
This is only the `Array: Module` conformance. Please add the `Array: Layer` conformance too, from https://github.com/tensorflow/swift-apis/issues/699.
```swift
extension Array: Layer where Element: Layer, Element.Input == Element.Output {}
```

Reply:
Marking as resolved, as this has now been added. Thanks @dan-zheng !